### PR TITLE
Migration to Angular 7 and RxJS 6

### PR DESCRIPTION
**Pull request by zabdalimovcsas**
Migration to Angular 7 and RxJS 6. Includes:

- Update of corresponding dependencies
- Peer dependencies versions changed
- Migration from .angular-cli.json to angular.json configuration file
- Build script updated according to new Angular cli
- Stronger typing of parameters in anonymous functions according to new typescript version
- RxJS 6 migration
  - Imports directly from 'rxjs' directory
  - Deprecated api is substituted by current one (functions over classes)
- Minor tslint, tsconfig and karma configuration file changes

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -10,7 +10,7 @@
     "start": "ng serve",
     "prebuild": "ng lint && ng test --watch=false",
```
Comment (mdvorak) [inline package.json]: Tests are now running twice, I'm pretty sure this wasn't the case with old config. Can be modified with --project param (for linter too I think, did not test).

File: src/app/dashboard/dashboard.component.ts [hunk 2]
```diff
@@ -14,6 +14,6 @@ export class DashboardComponent implements OnInit {
   }
 
   ngOnInit() {
-    this.view.body.subscribe(heroes => this.heroes = heroes);
+    this.view.body.subscribe((heroes: Heroes) => this.heroes = heroes);
```
Comment (mdvorak) [inline src/app/dashboard/dashboard.component.ts]: Why these changes? Applies to all components.
Generic signature is still automatic and linter is happy too. Unnescessary explicit code, used as an example.

File: src/app/hero-detail/hero-detail.component.ts [hunk 3]
```diff
@@ -17,7 +17,7 @@ export class HeroDetailComponent implements OnInit {
   }
 
   ngOnInit(): void {
-    this.view.body.subscribe(hero => this.hero = hero);
+    this.view.body.subscribe((hero: Hero) => this.hero = hero);
```
Comment (mdvorak) [inline src/app/hero-detail/hero-detail.component.ts]: As in dashboard.components.ts

File: src/app/heroes/heroes.component.ts [hunk 4]
```diff
@@ -16,7 +16,7 @@ export class HeroesComponent implements OnInit {
   }
 
   ngOnInit() {
-    this.view.body.subscribe(heroes => this.heroes = heroes);
+    this.view.body.subscribe((heroes: Heroes) => this.heroes = heroes);
```
Comment (mdvorak) [inline src/app/heroes/heroes.component.ts]: As in dashboard.components.ts

File: src/app/dashboard/dashboard.component.ts (hunk 2 above)
Comment (zabdalimovcsas) [reply src/app/dashboard/dashboard.component.ts]: I had feeling that there was build or lint errors because of that, but it's certantly not the case now - would switch it back to the original.

File: src/app/hero-detail/hero-detail.component.ts (hunk 3 above)
Comment (zabdalimovcsas) [reply src/app/hero-detail/hero-detail.component.ts]: See dashboard.components.ts comment.

File: src/app/heroes/heroes.component.ts (hunk 4 above)
Comment (zabdalimovcsas) [reply src/app/heroes/heroes.component.ts]: See dashboard.components.ts comment.

File: src/app/heroes/heroes.component.ts [hunk 5]
```diff
@@ -29,12 +29,20 @@ export class HeroesComponent implements OnInit {
 
     this.heroes.items.push(hero);
     this.http.post(this.heroes._links.self.href, hero)
-      .subscribe(undefined, undefined, () => this.view.reload());
+      .subscribe({
+        next: undefined,
```
[the remark this answers is not in the document]
Comment (zabdalimovcsas) [reply src/app/heroes/heroes.component.ts]: Got it, would just pass the arrow function as an argument then (although it becomes _next_ parameter instead of _complete_, but it doesn't affect the behavior, so i guess it's ok).

File: angular.json [hunk 6]
```diff
@@ -0,0 +1,269 @@
...
+              "e2e/tsconfig.e2e.json"
+            ],
+            "exclude": [
+              "**/node_modules/**"
+            ]
+          }
+        }
+      }
+    },
+    "angular-resource-router1": {
```
[the remark this answers is not in the document]
Comment (zabdalimovcsas) [reply angular.json]: It certainly does, but now there is no need for the second explicit config - builds with --prod flag are aot by default as i read. I deleted the second config ("angular-resource-router1" and corresponding e2e) for the application and keeped --prod flag in buildapp script for this purpose.

File: package.json (hunk 1 above)
Comment (zabdalimovcsas) [reply package.json]: You are right, it was because of --project param (and linting was performed 2 times as well). I deleted the param and corresponding project configuration (see angular.json comment).
